src/_pytest/config/__init__.py

```diff
@@ -656,7 +656,8 @@ def _importconftest(
         if dirpath in self._dirpath2confmods:
             for path, mods in self._dirpath2confmods.items():
                 if dirpath in path.parents or path == dirpath:
-                    assert mod not in mods
+                    error_message = f"{str(conftestpath)=} has already been loaded as the module {mod=}"
```

Comment:
I do think it will be useful to include the `__file__` as a debugging aid - it may no longer be at fault but it's useful to see where `mod` came from.

Reply:
Made this update